frontend/src/components/FieldList.tsx

```diff
@@ -34,6 +34,7 @@ export const FieldListField = ({
     <FormGroup isRequired={options.isRequired} label={options.label}>
       <ComponentField
         aria-label={`Field list ${options.key}`}
+        data-testid={options.key}
```

Comment:
Qualify this dynamic test ID:
```suggestion
        data-testid={`field ${options.key}`}
```